src/box/vy_log.c

```diff
@@ -1396,18 +1328,9 @@ void
 vy_log_write(const struct vy_log_record *record)
 {
 	say_verbose("write vylog record: %s", vy_log_record_str(record));
-
-	if (vy_log.tx_failed)
-		return;
-
 	assert(vy_log.tx != NULL);
 	struct vy_log_record *tx_record = vy_log_record_dup(&vy_log.tx->region,
```

Comment:
I would be nice to have it named `vy_log_record_xdup` or something to show it never fails, but I couldn't find anything similar in the vinyl and `xvy_log_record_dup` would look ugly.

The same note about `vy_recovery_alloc_key_parts` (e. g. `vy_recovery_xalloc_key_parts`).

Reply:
We don't use the `x` prefix with non-failing functions that allocate various objects, for example, see `key_def_dup`. The ultimate goal is to make all `alloc` and `dup` functions non-failing (perhaps, for a few rare exceptions, like memtx tuple allocations) so the `x` prefix would be pointless. The only reason why we added the `x` prefix to generic allocation functions, like `mempool_alloc` or `region_alloc`, is that we have a lot of legacy code handling errors returned by those functions, and the `x` prefix allows for a gradual transition.